nowcasting_dataset/data_sources/gsp/eso.py

```diff
@@ -105,10 +112,26 @@ def get_gsp_shape_from_eso(
         )
 
         with urlopen(url) as response:
-            shape_gpd = gpd.read_file(response).to_crs(WGS84_CRS)
+            shape_gpd = gpd.read_file(response)
+
+            # calculate the centroid before using - to_crs
+            shape_gpd["centroid_x"] = shape_gpd["geometry"].centroid.x
+            shape_gpd["centroid_y"] = shape_gpd["geometry"].centroid.y
+            shape_gpd["centroid_lat"], shape_gpd["centroid_lon"] = osgb_to_lat_lon(
+                x=shape_gpd["centroid_x"], y=shape_gpd["centroid_y"]
+            )
+
+            # project to WGS84
+            shape_gpd = shape_gpd.to_crs(WGS84_CRS)
```

Comment:
Is this still necessary?  I think the satellite data and the NWPs are in OSGB projection in the batches, right?  It'd be good to keep all the geospatial data in the same projection in the batches.  Or have I misunderstood?!